pages/library/_viewId.vue

```diff
@@ -81,6 +81,32 @@ export default Vue.extend({
       } else {
         return false;
       }
+    },
+    isDefaultView() {
+      const defaultViews = ['Series', 'Movie', 'Book', 'MusicAlbum'];
+
+      // Only purpose right now is to show proper text if filtering holds no results
+      // This could reside in onChangeFilter and onChangeType, but would require the same checks,
+      // better to have all of them in the same function I suppose
+      if (
+        this.statusFilter.length ||
+        this.genresFilter.length ||
+        this.yearsFilter.length ||
+        this.ratingsFilter.length ||
+        this.filterHasSubtitles ||
+        this.filterHasTrailer ||
+        this.filterHasSpecialFeature ||
+        this.filterHasThemeSong ||
+        this.filterHasThemeVideo ||
+        this.filterIsHd ||
+        this.filterIs4k ||
+        this.filterIs3d ||
+        !defaultViews.includes(this.viewType)
```

Comment:
Reviewing on phone so I might miss something but why do we need this?

Reply:
The default message is "The library is empty", even if it's not when applying filters.
This allows to display a better message when there is no result "No results found", which is way better.

Although we could write a poem:

> Your library's empty
> And your filters are cloudy,
> Thy shall review their search,
> So thee day can be sunny.

Although now I'm thinking we could save the inital number of results when they load then compare that with the current number of results.

When movies load set variable to 10, refine search no results, we compare 10 != 0, so it's a filter not a library... Could be better and we could display [2/31 movies] when selecting a genre for example

Reply:
But I don't get why we need to check which kind of item the grid is displaying (Music, Books...) which you defined in the const above

Reply:
@ferferga My bad, it's because we can 'sort' by collection, artist, etc. (Technically a _view_) If there is no collection in Movies it doesn't mean there is no movies, it means there are no collection.

Way later we could do something like this:
![Screenshot from 2020-12-01 13-16-11](https://user-images.githubusercontent.com/11581433/100780220-7642b300-33d7-11eb-8d54-47f88bd37283.png)
Make the message more friendly with `There is no collection`, etc with a link that could show tips.

Reply:
@Extarys Been a long time without running master so I missed that stuff, so thank you for explaining!

Reply:
Np, I thought you asked why we needed the function - I don't use git much (therefore not familiar with how PR works) I was confused.

Hope you laughed at the _poem_ ;)

Reply:
Also, I gave some thoughts at how I could avoid a function with the variable that keep the total number of movies in memory while applying filters, but when websocket comes or if we come to that page with filters pre-applied this will break so a function is a little bit future-proof.

Reply:
Some thoughts on this massive condition:

Wouldn't it be better to change the current message to "There is nothing here." (Effectively eliminating the need to check for the type of view we're in, thereby simplifying this a bit) as well as to move the check for filters to `onChangeFilter` ?

So we'd have a `data` named "hasFilters" or something, which would be set with the following (semi pseudo-code):
```typescript
onChangeFilter(filter: Record<string, any>) {
  this.hasFilters = Object.values(filter).every((value) => {
    return value.length > 0 || value !== false;
  })

  if (!hasFilters) return; // We don't have any filters, so no need to waste time doing more checks
}
```

This effectively remove the maintenance burden of having to change all the conditions manually if we ever add/remove filters in the future, while allowing us to keep the specific text for filters.